framework/OutStreams/PlotInterfaces/PlotInterface.py

```diff
@@ -0,0 +1,113 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on April 1, 2021
+
+@author: talbpaul
+"""
+
+from OutStreams import OutStreamInterface
+
+class PlotInterface(OutStreamInterface):
+  """
+    Archetype for Plot implementations
+  """
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for class "cls".
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for specifying the input of cls.
+    """
+    spec = super().getInputSpecification()
+    return spec
+
+  def __init__(self):
+    """
+      Init of Base class
+      @ In, None
+      @ Out, None
+    """
+    super().__init__()
+    self.printTag = 'PlotInterface'
+    self.overwrite = True     # overwrite existing creations? # TODO not in input specs?
+    self.subDirectory = None  # directory to save generated files to # TODO should be in spec
+    self.filename = ''        # target file name? TODO should be in spec
+
+  def handleInput(self, spec):
+    """
+      Loads the input specs for this object.
+      @ In, spec, InputData.ParameterInput, input specifications
+      @ Out, None
+    """
+    super().handleInput(spec)
+
+  def initialize(self, inDict):
+    """
+      Function to initialize the OutStream. It basically looks for the "data"
+      object and links it to the system.
+      @ In, inDict, dict, contains all the Objects are going to be used in the
+      current step. The sources are searched into this.
+      @ Out, None
+    """
+    super().initialize(inDict)
+
+  def run(self):
+    """
+      Main run method.
+      @ In, None
+      @ Out, None
+    """
+    pass
+
+  ##################
+  # Utility
+  def findSource(self, name, stepEntities):
+    """
+      Find a source from the potential step sources.
+      @ In, name, str, name of the source
+      @ In, stepEntities, dict, entities from the Step
+      @ Out, findSource, object, discovered object or None
+    """
+    for out in stepEntities['Output']:
+      if out.name == name:
+        return out
+    for inp in stepEntities['Input']:
+      if inp.name == name:
+        return inp
+    for other in ['TargetEvaluation', 'SolutionExport']:
+      if other in stepEntities:
+        if stepEntities[other].name == name:
+          return stepEntities[other]
+    return None
+
+  def getInitParams(self):
```

Comment:
This method is overwritten, please check if this is what you want.

Reply:
the method doesn't exist in the base class, and according to @alfoa needs to be removed generally. I will add a passthrough call to the base class for comprehensiveness.

Reply:
no the getInitiParams will stay. The getCurrentParams will be removed